domain/consensus/processes/pruningmanager/pruningmanager.go

```diff
@@ -221,47 +224,47 @@ func (pm *pruningManager) deletePastBlocks(pruningPoint *externalapi.DomainHash)
 	onEnd := logger.LogAndMeasureExecutionTime(log, "pruningManager.deletePastBlocks")
 	defer onEnd()
 
-	// Go over all P.Past and P.AC that's not in V.Past
+	// Go over all pruningPoint.Past and pruningPoint.Anticone that's not in virtual.Past
 	queue := pm.dagTraversalManager.NewDownHeap()
-
-	// Find P.AC that's not in V.Past
-	dagTips, err := pm.consensusStateStore.Tips(pm.databaseContext)
+	virtualParents, err := pm.dagTopologyManager.Parents(model.VirtualBlockHash)
 	if err != nil {
 		return err
 	}
-	newTips := make([]*externalapi.DomainHash, 0, len(dagTips))
-	virtualParents, err := pm.dagTopologyManager.Parents(model.VirtualBlockHash)
+
+	// Start queue with all tips that are below the pruning point (and on the way remove them from list of tips)
+	prunedTips, err := pm.pruneTips(pruningPoint, virtualParents)
 	if err != nil {
 		return err
 	}
-	for _, tip := range dagTips {
-		isInPruningFutureOrInVirtualPast, err := pm.isInPruningFutureOrInVirtualPast(tip, pruningPoint, virtualParents)
-		if err != nil {
-			return err
-		}
-		if !isInPruningFutureOrInVirtualPast {
-			// Add them to the queue so they and their past will be pruned
-			err := queue.Push(tip)
-			if err != nil {
-				return err
-			}
-		} else {
-			newTips = append(newTips, tip)
-		}
+	err = queue.PushSlice(prunedTips)
+	if err != nil {
+		return err
 	}
-	pm.consensusStateStore.StageTips(newTips)
-	// Add P.Parents
+
+	// Add pruningPoint.Parents to queue
 	parents, err := pm.dagTopologyManager.Parents(pruningPoint)
 	if err != nil {
 		return err
 	}
-	for _, parent := range parents {
-		err = queue.Push(parent)
-		if err != nil {
-			return err
-		}
+	err = queue.PushSlice(parents)
+	if err != nil {
+		return err
+	}
+
+	err = pm.deleteBlocksDownward(queue)
+	if err != nil {
+		return err
+	}
+
+	err = pm.pruneVirtualDiffParents(pruningPoint, virtualParents)
+	if err != nil {
+		return err
 	}
 
+	return nil
+}
+
+func (pm *pruningManager) deleteBlocksDownward(queue model.BlockHeap) error {
```

Comment:
Can you repalce the parents loop in there(line 286-287) with `queue.PushSlice(parents)`?